velox/expression/Expr.cpp

```diff
@@ -102,6 +102,18 @@ bool hasConditionals(Expr* expr) {
   return false;
 }
 
+void findMultiRefFields(
+    std::set<FieldReference*>& allFields,
+    std::set<FieldReference*>& multiRefFields,
+    const std::vector<FieldReference*>& moreFields) {
+  for (auto* newField : moreFields) {
+    if (allFields.find(newField) != allFields.end()) {
+      multiRefFields.insert(newField);
+    }
+    allFields.insert(newField);
```

Comment:
nit: This logic can be moved inside the 'if' statement. If allFields already contains the newField, then there is no point in inserting it.